src/builtins.ts

```diff
@@ -2997,7 +2997,7 @@ function builtin_function_call(ctx: BuiltinContext): ExpressionRef {
     compiler.currentType = returnType;
     return compiler.module.unreachable();
   }
-  var indexArg = compiler.compileExpression(assert(ctx.thisOperand), ftype, Constraints.CONV_IMPLICIT);
+  var functionInstance = compiler.compileExpression(assert(ctx.thisOperand), ftype, Constraints.CONV_IMPLICIT);
```

Comment:
We are using `functionInstance` at other places to indicate a `Function`, so perhaps `functionPointer`?

Reply:
Or maybe, to stick to other `expr` names, `functionPointerExpr` and `tableIndexExpr` respectively.